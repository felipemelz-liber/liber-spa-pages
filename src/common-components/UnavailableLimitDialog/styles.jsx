import styled from 'styled-components';
import { FlatButton } from 'liber-components';

const TEXT_COLOR = '#6287a7';

export const Container = styled.div`
  padding: 24px 24px 12px 24px;
  width: 480px;

  * {
    font-family: Roboto;
  }
`;

export const Header = styled.div`
  font-size: 21px;
  font-weight: bold;
  line-height: 1.4;
  letter-spacing: 0.08px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const Content = styled.div`
  font-size: 16px;
  line-height: 1.38;
  letter-spacing: 0.01px;
  margin-bottom: 24px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const ActionsRow = styled.div`
  display: flex;
  justify-content: flex-end;
`;

export const Button = styled(FlatButton).attrs(() => ({ size: 'large' }))``;
