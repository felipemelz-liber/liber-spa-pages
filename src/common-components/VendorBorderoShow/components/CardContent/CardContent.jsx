import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Icon from '@mdi/react';
import { mdiDownload, mdiCheckboxMarkedCircle, mdiClock } from '@mdi/js';
import {
  Row,
  Title,
  FlatButton,
  FieldContainer,
  FieldLabel,
  FieldValue,
  StatusIcon,
  Caption,
} from './CardContent.styles';
import BankAccount from '../BankAccount/BankAccount';
import { dispatchDataLayerEvent } from '../../../../vendor/Utils';

const CardContent = ({ summary }, isAdmin) => {
  const {
    buyer: { name: buyerName, cnpj },
    paidValue,
    netValue,
    invoiceCount,
    paidInvoiceCount,
    requestedAt,
    operators,
    bankAccount,
    signatureCertificateUrl,
    invoiceTransferAgreementUrl,
    signatures: { required },
  } = summary;

  const handleOpenDocument = url => {
    window.open(url);
  };

  const handleClickSigningCertificate = () => {
    dispatchDataLayerEvent('showBorderoBaixarCertificadoDeAssinatura', isAdmin);

    handleOpenDocument(signatureCertificateUrl);
  };

  const handleClickTransferAgreement = () => {
    dispatchDataLayerEvent('showBorderoBaixarTermoDeCessao', isAdmin);

    handleOpenDocument(invoiceTransferAgreementUrl);
  };

  const needConfirmation = () =>
    operators.reduce((prev, { userHasSigned }) => prev || !userHasSigned, false);

  const getSubstantives = number => `representante${number > 1 ? 's legais' : ' legal'}`;

  const getPaidValue = value => {
    return value === 'R$ 0,00' ? '***' : value;
  };

  return (
    <>
      <Row justify="space-between">
        <Title>Resumo</Title>
        <div>
          {signatureCertificateUrl && (
            <FlatButton blue onClick={handleClickSigningCertificate}>
              <Icon path={mdiDownload} />
              Certificado de Assinatura
            </FlatButton>
          )}
          {invoiceTransferAgreementUrl && (
            <FlatButton blue paddingRight="0" onClick={handleClickTransferAgreement}>
              <Icon path={mdiDownload} />
              Termo de Cessão
            </FlatButton>
          )}
        </div>
      </Row>
      <Row height={24} />
      <Row>
        <FieldContainer>
          <FieldLabel>Sacado</FieldLabel>
          <FieldValue>{`${buyerName} — CNPJ ${cnpj}`}</FieldValue>
        </FieldContainer>
      </Row>
      <Row height={32} />
      <Row>
        <FieldContainer marginRight={60}>
          <FieldLabel>Valor {needConfirmation() ? 'Solicitado' : 'Antecipado'}</FieldLabel>
          <FieldValue>{needConfirmation() ? netValue : getPaidValue(paidValue)}</FieldValue>
        </FieldContainer>
        <FieldContainer marginRight={60}>
          <FieldLabel>Títulos Solicitados</FieldLabel>
          <FieldValue>{invoiceCount}</FieldValue>
        </FieldContainer>
        <FieldContainer marginRight={60}>
          <FieldLabel>Títulos Antecipados</FieldLabel>
          <FieldValue>{paidInvoiceCount || '***'}</FieldValue>
        </FieldContainer>
        <FieldContainer>
          <FieldLabel>Solicitação</FieldLabel>
          <FieldValue>{requestedAt}</FieldValue>
        </FieldContainer>
      </Row>
      <Row height={52} />
      <Row>
        <Title>Confirmação</Title>
        {needConfirmation() && (
          <Caption>
            {`*Esta antecipação exige a confirmação de ${required} ${getSubstantives(required)}`}
          </Caption>
        )}
      </Row>
      <Row height={22} />
      {operators.map(({ id, name, userHasSigned }) => (
        <Row key={`operator-${id}`}>
          <StatusIcon
            status={userHasSigned ? 'positive' : 'neutral'}
            path={userHasSigned ? mdiCheckboxMarkedCircle : mdiClock}
          />
          <FieldValue>{name}</FieldValue>
        </Row>
      ))}
      <Row height={48} />
      <Title>Pagamento</Title>
      <Row height={24} />
      <Row>
        <BankAccount {...bankAccount} />
      </Row>
    </>
  );
};

CardContent.propTypes = {
  summary: PropTypes.shape({
    buyer: PropTypes.shape({
      name: PropTypes.string,
      cnpj: PropTypes.string,
    }),
    paidValue: PropTypes.string,
    netValue: PropTypes.string,
    invoiceCount: PropTypes.string,
    paidInvoiceCount: PropTypes.string,
    requestedAt: PropTypes.string,
    operators: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number,
        name: PropTypes.string,
        userHasSigned: PropTypes.bool,
      }),
    ),
    bankAccount: {
      branch: PropTypes.string,
      branchCd: PropTypes.string,
      account: PropTypes.string,
      accountCd: PropTypes.string,
      bankName: PropTypes.string,
      id: PropTypes.number,
      bankCode: PropTypes.string,
    },
    signatureCertificateUrl: PropTypes.string,
    invoiceTransferAgreementUrl: PropTypes.string,
    signatures: PropTypes.shape({
      required: PropTypes.number,
    }),
  }),
  isAdmin: PropTypes.bool,
};

CardContent.defaultProps = {
  summary: {
    buyer: '',
    paidValue: '',
    netValue: '',
    invoiceCount: '',
    paidInvoiceCount: '',
    requestedAt: '',
    operators: [],
    bankAccount: {
      branch: '',
      branchCd: '',
      account: '',
      accountCd: '',
      bankName: '',
      id: null,
      bankCode: '',
    },
    signatureCertificateUrl: '',
    invoiceTransferAgreementUrl: '',
    signatures: {
      required: 1,
    },
  },
  isAdmin: false,
};

const mapStateToProps = ({ vendorBorderoShow: { summary } }) => ({ summary });

export default connect(mapStateToProps)(CardContent);
