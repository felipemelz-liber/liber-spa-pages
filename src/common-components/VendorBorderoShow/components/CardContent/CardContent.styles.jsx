import styled, { css } from 'styled-components';
import { Button } from 'liber-components';
import Icon from '@mdi/react';

const colorReference = {
  positive: '#008a5e',
  negative: '#ff5b56',
  neutral: '#6287a7',
};

export const Row = styled.div`
  width: 100%;
  display: flex;
  align-items: center;

  ${({ height }) =>
    height &&
    css`
      height: ${height}px;
    `}

  ${({ justify }) =>
    justify &&
    css`
      justify-content: ${justify};
    `}
`;

export const Title = styled.div`
  font-family: Roboto;
  font-size: 21px;
  font-weight: 500;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.4;
  letter-spacing: 0.08px;
  text-align: left;
  color: #4b6f85;
`;

export const FieldLabel = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: 500;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.4;
  letter-spacing: 0.06px;
  text-align: left;
  color: #405f71;
`;

export const FieldValue = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.84;
  letter-spacing: 0.06px;
  text-align: left;
  color: #6287a7;
`;

export const FlatButton = styled(Button).attrs(() => ({ version: 2 }))`
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.25;
  letter-spacing: 0.02px;
  text-align: right;
  color: ${({ blue }) => (blue ? '#009dff' : '#798e9b')};

  & > svg {
    fill: ${({ blue }) => (blue ? '#009dff' : '#798e9b')};
    width: 20px;
    height: 20px;
    margin-right: 8px;
  }

  ${({ marginRight }) =>
    marginRight &&
    css`
      margin-right: ${marginRight}px;
    `}

  ${({ paddingRight }) =>
    paddingRight &&
    css`
      padding-right: ${paddingRight}px;
    `}
`;

export const FieldContainer = styled.div`
  height: 46px;

  ${({ marginRight }) =>
    marginRight &&
    css`
      margin-right: ${marginRight}px;
    `}
`;

export const StatusIcon = styled(Icon)`
  width: 16px;
  height: 16px;
  margin-right: 8px;
  margin-bottom: 2px;
  fill: ${({ status = 'neutral' }) => colorReference[status]};
`;

export const Caption = styled.div`
  font-family: Roboto;
  font-size: 12px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  letter-spacing: 0.05px;
  color: #6287a7;
  margin: 0px 16px;
`;
