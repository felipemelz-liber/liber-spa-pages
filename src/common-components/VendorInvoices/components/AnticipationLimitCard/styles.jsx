import styled from 'styled-components';

export const Card = styled.div`
  font-family: Roboto;
  color: #6287a7;
  width: 360px;
  box-shadow: 0px 3px 6px #00000029;
  border: 1px solid #cdd7df;
  border-radius: 8px;
  padding: 16px 24px;
  margin-bottom: 56px;

  * {
    font-family: Roboto;
  }
`;

export const CardTitle = styled.div`
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.06px;
  font-weight: 500;
  color: #4b6f85;
  margin-bottom: 4px;
`;
