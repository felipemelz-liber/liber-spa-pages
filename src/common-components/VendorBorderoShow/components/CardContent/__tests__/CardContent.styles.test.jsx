import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import * as Components from '../CardContent.styles';

describe('CardContent styled components test', () => {
  configure({ adapter: new Adapter() });

  Object.keys(Components).forEach(componentName => {
    const Component = Components[componentName];
    it(`should render ${componentName} correctly`, () => {
      const wrapper = shallow(<Component />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  const { FlatButton } = Components;
  it('should render FlatButton with blue property correctly', () => {
    const wrapper = shallow(<FlatButton blue />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  const { StatusIcon } = Components;
  it('should render StatusIcon with custom color correctly', () => {
    const wrapper = shallow(<StatusIcon color="#009dff" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
