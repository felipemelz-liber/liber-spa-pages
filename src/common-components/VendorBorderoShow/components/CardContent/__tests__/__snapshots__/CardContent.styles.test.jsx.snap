// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContent styled components test should render Caption correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-gzVnrw",
        "isStatic": false,
        "rules": Array [
          "
  font-family: Roboto;
  font-size: 12px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  letter-spacing: 0.05px;
  color: #6287a7;
  margin: 0px 16px;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-gzVnrw",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`CardContent styled components test should render FieldContainer correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-EHOje",
        "isStatic": false,
        "rules": Array [
          "
  height: 46px;

  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-EHOje",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`CardContent styled components test should render FieldLabel correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": false,
        "rules": Array [
          "
  font-family: Roboto;
  font-size: 16px;
  font-weight: 500;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.4;
  letter-spacing: 0.06px;
  text-align: left;
  color: #405f71;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`CardContent styled components test should render FieldValue correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": false,
        "rules": Array [
          "
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.84;
  letter-spacing: 0.06px;
  text-align: left;
  color: #6287a7;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`CardContent styled components test should render FlatButton correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [
        [Function],
      ],
      "componentStyle": ComponentStyle {
        "componentId": "sc-ifAKCX",
        "isStatic": false,
        "rules": Array [
          "
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.25;
  letter-spacing: 0.02px;
  text-align: right;
  color: ",
          [Function],
          ";

  & > svg {
    fill: ",
          [Function],
          ";
    width: 20px;
    height: 20px;
    margin-right: 8px;
  }

  ",
          [Function],
          "

  ",
          [Function],
          "
",
        ],
      },
      "displayName": "Styled(Buttons__Button)",
      "foldedComponentIds": Array [
        "Buttons__Button-u35aut-0",
      ],
      "render": [Function],
      "styledComponentId": "sc-ifAKCX",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`CardContent styled components test should render FlatButton with blue property correctly 1`] = `
<StyledComponent
  blue={true}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [
        [Function],
      ],
      "componentStyle": ComponentStyle {
        "componentId": "sc-ifAKCX",
        "isStatic": false,
        "rules": Array [
          "
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.25;
  letter-spacing: 0.02px;
  text-align: right;
  color: ",
          [Function],
          ";

  & > svg {
    fill: ",
          [Function],
          ";
    width: 20px;
    height: 20px;
    margin-right: 8px;
  }

  ",
          [Function],
          "

  ",
          [Function],
          "
",
        ],
      },
      "displayName": "Styled(Buttons__Button)",
      "foldedComponentIds": Array [
        "Buttons__Button-u35aut-0",
      ],
      "render": [Function],
      "styledComponentId": "sc-ifAKCX",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`CardContent styled components test should render Row correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  width: 100%;
  display: flex;
  align-items: center;

  ",
          [Function],
          "

  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`CardContent styled components test should render StatusIcon correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bZQynM",
        "isStatic": false,
        "rules": Array [
          "
  width: 16px;
  height: 16px;
  margin-right: 8px;
  margin-bottom: 2px;
  fill: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "Styled(Icon)",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "className": [Function],
        "color": [Function],
        "horizontal": [Function],
        "inStack": [Function],
        "path": [Function],
        "rotate": [Function],
        "size": [Function],
        "spin": [Function],
        "style": [Function],
        "vertical": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bZQynM",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "color": "currentColor",
          "horizontal": false,
          "rotate": 0,
          "size": null,
          "spin": false,
          "vertical": false,
        },
        "displayName": "Icon",
        "propTypes": Object {
          "className": [Function],
          "color": [Function],
          "horizontal": [Function],
          "inStack": [Function],
          "path": [Function],
          "rotate": [Function],
          "size": [Function],
          "spin": [Function],
          "style": [Function],
          "vertical": [Function],
        },
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`CardContent styled components test should render StatusIcon with custom color correctly 1`] = `
<StyledComponent
  color="#009dff"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bZQynM",
        "isStatic": false,
        "rules": Array [
          "
  width: 16px;
  height: 16px;
  margin-right: 8px;
  margin-bottom: 2px;
  fill: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "Styled(Icon)",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "className": [Function],
        "color": [Function],
        "horizontal": [Function],
        "inStack": [Function],
        "path": [Function],
        "rotate": [Function],
        "size": [Function],
        "spin": [Function],
        "style": [Function],
        "vertical": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bZQynM",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "color": "currentColor",
          "horizontal": false,
          "rotate": 0,
          "size": null,
          "spin": false,
          "vertical": false,
        },
        "displayName": "Icon",
        "propTypes": Object {
          "className": [Function],
          "color": [Function],
          "horizontal": [Function],
          "inStack": [Function],
          "path": [Function],
          "rotate": [Function],
          "size": [Function],
          "spin": [Function],
          "style": [Function],
          "vertical": [Function],
        },
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`CardContent styled components test should render Title correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  font-family: Roboto;
  font-size: 21px;
  font-weight: 500;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.4;
  letter-spacing: 0.08px;
  text-align: left;
  color: #4b6f85;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
